js/src/pages/edit-paid-ads-campaign/index.js

```diff
@@ -48,6 +51,15 @@ function getCurrentStep() {
 	return STEP.CAMPAIGN;
 }
 
+function isNotOurStep( location ) {
+	const allowList = new Set( [
+		getNewPath( { step: STEP.CAMPAIGN } ),
+		getNewPath( { step: STEP.ASSET_GROUP } ),
+	] );
+	const destination = location.pathname + location.search;
+	return ! allowList.has( destination );
```

Comment:
Since the different orders of `location.search` are still equivalent and this comparison process doesn't guarantee order when forming `location.search` strings, this determination might not be reliable enough across all JS engines.

Although I haven't encountered related issues when testing, I would recommend considering a more stable comparison method.

Reply:
Thanks @eason9487 Afraid my knowledge in JS is not wide to find more stable methods. But I will investigate further and I will try to make it to happen in follow up PRs. 

Feel free to suggest what do you think would be the best approach!